### DISCUSSION TITLE
Launchpad replacement: Start menu

### DISCUSSION THREAD

**Discussion by lawand-dot-io** [lawand-dot-io/taskbar#31] [Feature requests]
Show a windows-style start menu instead of the launchpad

(Feature request split from https://github.com/lawand-dot-io/taskbar/discussions/25)

**Comment by kjp2006** [comment]
I would also add that if you add a start menu, is it possible to create a sub menu for Last used, or and area to pin those apps to for quick access? If you need a coding reference point, check out Stardock for mac. they have a win10 version that's not bulky for mac as well as "start menu for mac" though I love this minimal high feature design of Taskbar and it seems waay more promising than the others comparatively.

**Comment by lawand-dot-io** [comment]
@kjp2006 I'm glad you enjoy Taskbar. I will keep your suggestions in mind. I saw this mockup recently so I'll attach it here for future reference. This is the ideal start menu (and includes the details you mentioned), but I'm not sure how much work it will take for me to implement. Let's see

<img width="873" alt="Screenshot 2024-06-23 at 01 39 17" src="https://github.com/lawand-dot-io/taskbar/assets/132961914/8b8df6cd-4f14-49d8-a7d8-a08e7ec7363e">

**Reply by caliclone** [reply]
Taskbar is great! For me, the only thing missing is a start menu. I'd pay good money for this product if a start menu was included. Such excellent work so far. Thank you!!

**Reply by lawand-dot-io** [reply]
@caliclone Thank you :)

**Comment by shsn** [comment]
I don’t necessarily need a full Start menu, but I’d love to see a customizable icon or image on the left side of the menu—something I can fully personalize, both in appearance and function. For me, the real value would be in the flexibility: being able to assign it to launch Spotlight, Raycast (a Spotlight alternative), or even open a custom list of apps, files, or resources.

I’m less interested in replicating a traditional Start menu and more excited by the potential for creative and personalized use cases.

Just wanted to add my voice to the enthusiasm around this feature request—even if my ideal setup might be a bit different from others.

**Reply by lawand-dot-io** [reply]
@shsn Thanks for the suggestion. I moved your comment to a new request (https://github.com/lawand-dot-io/taskbar/discussions/169) to keep it separate :) cheers

**Reply by lawand-dot-io** [reply]
By the way, if you're still interested, I realized that it's already possible to launch raycast or other apps instead of the launchpad. all you have to do is to hide the launchpad icon in preferences, then pin the other app (such as Raycast) using the "Keep in Taskbar" option

**Comment by lawand-dot-io** [comment]
Update: I am working on implementing this at the moment. This is how it looks like so far. The development is almost done. I might add a text search option. Hope this will be a useful feature :)
<img width="2560" height="1450" alt="Startmenu-prototype" src="https://github.com/user-attachments/assets/722262d0-9f22-471d-bd57-62faca502813" />

**Comment by kjp2006** [comment]
Very very nice and impressive. I don’t know if opacity sliders integrated
into this would be easy to implement into settings with this but this
already looks like the best taskbar for Mac so far.

**Reply by lawand-dot-io** [reply]
Thanks :) yeah, I am aware some users prefer more transparency, this will be a separate issue. The start menu already works well, I will test it a bit more then release it :)

**Comment by lawand-dot-io** [comment]
Hello. If anybody is interested in beta testing the start menu release, send me an email:  info@lawand.io

Thanks

**Comment by lawand-dot-io** [comment]
The new Taskbar version with a start menu and transparency option has slowed progress due to a health issue, but development continues. You can try the latest, mostly stable but not yet fully tested version via [this link](https://lawand.io/wp-content/uploads/2025/11/taskbar-1.5.3.zip). Please report any bugs, thanks

If this version works without issues, you can continue using it  and once I make an official release you'll see an option to update.

**Comment by pulpdood** [comment]
HI Lawand, just a small feature request, is it possible to have a keyboard shortcut to open the start menu? Thank you :)

Also, I have a weird display bug:

<img width="418" height="315" alt="image" src="https://github.com/user-attachments/assets/eda5c420-d7a0-4c06-97bf-00799fca591c" />

Sorry to hear about the health issue. I hope you get well soon!

**Reply by lawand-dot-io** [reply]
@pulpdood hello :) there is already request for that (https://github.com/lawand-dot-io/taskbar/discussions/270) but I'm not sure if I will do it because it's tricky for my app to get keyboard shortcuts since that will Break its behavior a little bit because it's supposed to be always visible and never receive keyboard events.

As for the bug you mentioned, does this always happen? Does it go away when restarting Taskbar?

And also thank you for the well wishes, my health is gradually improving so hopefully I will get back to working on taskbar soon :)

**Reply by pulpdood** [reply]
@lawand-dot-io Hi Lawand, ah all good, don't want to overcomplicate the logic. 

The bug is always there even when restarting Taskbar and restarting macOS. Also, have updated macOS to latest version and issue persists.

Glad to hear you're getting better :)

**Comment by lawand-dot-io** [comment]
This is now implemented in v1.6.1 which is now released
